oh.ps1

```diff
@@ -810,211 +795,263 @@ cd "$OH_PATH" # workaround for hard coded paths
-		        	Write-Host "Error: no data found! Exiting." -ForegroundColor Red
-				Read-Host; exit 1
+		###################################################
+		"P"	{ # start in PORTABLE mode
+			$script:OH_MODE="PORTABLE"
+			Write-Host "OH_MODE set to PORTABLE mode." -ForeGroundcolor Green
+			Read-Host;
+		}
+		###################################################
+		"S"	{ # start in SERVER (Portable) mode
+			$script:OH_MODE="SERVER"
+			Write-Host "OH_MODE set to SERVER mode." -ForeGroundcolor Green
+			Read-Host;
+		}
+		###################################################
+		"d"	{ # debug 
+			$script:LOG_LEVEL="DEBUG"
+			Write-Host "Log level set to $LOG_LEVEL"
+			Read-Host;
+		}
+		###################################################
+		"D"	{ # demo mode 
+			# exit if OH is configured in CLIENT mode
+			if ( $OH_MODE -eq "CLIENT" ) {
+				Write-Host "Error - OH_MODE set to CLIENT mode. Cannot run with Demo data." -ForeGroundcolor Red
+				Read-Host;
 			}
+			else { $script:OH_MODE="PORTABLE" }
+			$DEMO_DATA="on"
+			Write-Host "Demo data set to on. Using demo data."
+			Read-Host;
 		}
-		test_database_connection;
-		Write-Host "Saving Open Hospital database..."
-		dump_database;
-		if ( $OH_MODE -eq "PORTABLE" ) {
-			shutdown_database;
+		###################################################
+		"g"	{ # generate config files and exit
+			$script:GENERATE_CONFIG_FILES="on"
+			generate_config_files;
+			Write-Host "Done!"
+			Read-Host;
 		}
-		Write-Host "Done!"
-		Read-Host;
-		exit 0
-	}
-	"r"	{ # restore
-	       	Write-Host "Restoring Open Hospital database...."
-		# ask user for database to restore
-		$DB_CREATE_SQL = Read-Host -Prompt "Enter SQL dump/backup file that you want to restore - (in $script:SQL_DIR subdirectory) -> "
-		if ( !(Test-Path "$OH_PATH\$SQL_DIR\$DB_CREATE_SQL")) {
-			Write-Host "Error: No SQL file found! Exiting." -ForegroundColor Red
-			Read-Host; exit 2
+		###################################################
+		"G"	{ # set up GSM 
+			Write-Host "Setting up GSM..."
+			java_check;
+			java_lib_setup;
+			Start-Process -FilePath "$JAVA_BIN" -ArgumentList ("-Djava.library.path=${NATIVE_LIB_PATH} -classpath $OH_CLASSPATH org.isf.utils.sms.SetupGSM $@ ") -Wait -NoNewWindow
+			Write-Host "Done!"
+			Read-Host;
+		}
+		###################################################
+		"i"	{ # initialize/install OH database
+			# set mode to CLIENT
+			$OH_MODE="CLIENT"
+			Write-Host "Do you want to initialize/install the OH database on:"
+			Write-Host ""
+			Write-Host " Server -> $DATABASE_SERVER"
+			Write-Host " TCP port -> $DATABASE_PORT"
+			Write-Host ""
+			get_confirmation;
+			set_language;
+			initialize_dir_structure;
+			mysql_check;
+			# ask user for database root password
+			$script:DATABASE_ROOT_PW = Read-Host "Please insert the MySQL / MariaDB database root password (root@$DATABASE_SERVER) -> "
+			Write-Host "Installing the database....."
+			Write-Host ""
+			Write-Host " Database name -> $DATABASE_NAME"
```

Comment:
Small nitpick: Would it make sense to write out the just entered $DATABASE_ROOT_PW to ensure they really entered what they thought they entered?  So later if they want to access the DB as root they really know what the password is.

Reply:
It's actually considered a bad practice: Even Powershell comments it out...